application/src/main/java/org/thingsboard/server/service/security/auth/oauth2/AbstractOAuth2ClientMapper.java

```diff
@@ -151,6 +156,9 @@ protected SecurityUser getOrCreateSecurityUserFromOAuth2User(OAuth2User oauth2Us
                         UserCredentials userCredentials = userService.findUserCredentialsByUserId(user.getTenantId(), user.getId());
                         userService.activateUserCredentials(user.getTenantId(), userCredentials.getActivateToken(), passwordEncoder.encode(""));
                     }
+
+                    notificationEntityService.notifyCreateOrUpdateOrDelete(tenantId, customerId, user.getId(),
```

Comment:
Why not to use TbUserService ?

Reply:
Because we need to have an user id who has created another user to use TbUserService.
In case of Oauth2 we do not have a such id.